### DISCUSSION TITLE
"Ready" in real-time multiplayer lobbies still allows you to change critical settings

### DISCUSSION THREAD

**Discussion by zachmanthethird** [ppy/osu#16553] [Q&A]
1. Players can change their team color while being readied. (https://youtu.be/DDVeIYY3pYw?t=2205) This allows players to swap teams right as the match is started, and you can see in the video that this causes some discrepancies on who is on which team.
2. Players can change their FreeMod mods while being readied. (https://youtu.be/DDVeIYY3pYw?t=2655) While less crucial, this allows players to select/deselect mods possibly required for tournament play.
3. The host can change the map while still being readied. (https://youtu.be/DDVeIYY3pYw?t=5910) While not tested with other players in the lobby, a map change should make all players unready to receive additional consent before the map starts.

Lazer Version 2022.108.0

**Comment by bdach** [comment]
> While less crucial, this allows players to select/deselect mods possibly required for tournament play.

If the mod is required then it shouldn't be a free mod in the first place, I would think?

But generally I agree that in all of the scenarios presented it sounds reasonable to unready the user(s).

**Reply by dnialh** [reply]
Some current free mod rules that I don't think can be forced in current multi-player would be 'You must play at least one of hidden or hard rock' and 'At least one player per team must use hidden'.

Both of these rules require people to select their own mods using free mod.

**Reply by bdach** [reply]
I see, that makes sense. Not sure if those use cases are going to get targeted support as they seem a little specific but maybe?

**Reply by peppy** [reply]
I don't see this as a critical issue for one reason: ready does not lock you in permanently. A user could just as easily make a clicking macro to quickly unready, change the mods, and ready again.

If this is seen as an issue then we need a mode where once locked in, a user cannot un-ready, or requires a referee to reset their status.

I'm neutral on whether being "ready" should lock options or not.
